### PR TITLE
Add masked similarity computation and fjaccard

### PR DESCRIPTION
**Pull request by koheiw**
This is a branch for #58.  Using `mask`, we can compute only the similarity between columns with the same names, for example.

``` r
require(proxyC)
#> Loading required package: proxyC
#> 
#> Attaching package: 'proxyC'
#> The following object is masked from 'package:stats':
#> 
#>     dist
mt1 <- Matrix::rsparsematrix(100, 6, 1.0)
colnames(mt1) <- c("a", "a", "d", "d", "e", "e")
mt2 <- Matrix::rsparsematrix(100, 5, 1.0)
colnames(mt2) <- c("a", "b", "c", "d", "e")

(msk <- mask(colnames(mt1), colnames(mt2)))
#> 6 x 5 sparse Matrix of class "lgTMatrix"
#>   a b c d e
#> a | . . . .
#> a | . . . .
#> d . . . | .
#> d . . . | .
#> e . . . . |
#> e . . . . |
simil(mt1, mt2, margin = 2, mask = msk, drop0 = TRUE)
#> 6 x 5 sparse Matrix of class "dgTMatrix"
#>            a b c           d          e
#> a 0.02872607 . .  .          .         
#> a 0.02847397 . .  .          .         
#> d .          . . -0.05835794 .         
#> d .          . .  0.03717545 .         
#> e .          . .  .          0.06007289
#> e .          . .  .          0.09538591
```

<sup>Created on 2025-04-17 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

### CODE REVIEW

File: tests/testthat/test-mask.R [hunk 1]
```diff
@@ -0,0 +1,119 @@
+require(Matrix)
```
Comment (rcannood) [inline tests/testthat/test-mask.R:1]: Can we also test whether simil() is equivalent when `mask = NULL` vs. when `mask = mask(c("a", "a", "a", "a", ...), c("a", "a", "a", "a", ...))` ?

**Review by rcannood — COMMENTED** [review]
Oh, great idea, I can imagine this is really useful in some use cases!

Could you add a section in one of the vignettes demonstrating its functionality?

I don't expect this PR to affect the performance, but if we rerun the readme, is there any change in performance?

File: tests/testthat/test-mask.R (hunk 1 above)
Comment (koheiw) [reply tests/testthat/test-mask.R:1]: Thanks. I added tests to compare `mask = NULL` and `mask` is matrix with all `TRUE`.

Comment (koheiw) [reply tests/testthat/test-mask.R:1]: Also added a vignette page. Please feel free to edit.

**Comment by koheiw** [comment]
Should `simil()` return `NaN` or `NA` for masked cells?

File: tests/testthat/test-mask.R (hunk 1 above)
Comment (rcannood) [reply tests/testthat/test-mask.R:1]: Thanks! :bow:

**Comment by rcannood** [comment]
> Should `simil()` return `NaN` or `NA` for masked cells?

I would indeed expect NAs ok masked cells, even if that's a bit unfortunate in terms of data storage 😅

**Review by rcannood — APPROVED** [review]
LGTM! Unless there are more things you'd like to see changed?

**Comment by koheiw** [comment]
I think I should add `mask` to `dist()`. It is possible because masked cells are now NaN.

**Review by rcannood — CHANGES_REQUESTED** [review]
Good point! In that case, let me know when you'd like a rereview!

**Comment by koheiw** [comment]
@rcannood  `dist() has mask now. If you think its fine, please merge this PR.

**Comment by rcannood** [comment]
Could you merge the master branch into this one? 🙇

File: docs/404.html
Comment (rcannood) [inline docs/404.html:1]: It's better not to update the docs manually, as this could lead to unnecessary conflicts or incorrect/outdated docs, but we can solve this in a next PR.

See #62

**Review by rcannood — APPROVED** [review]
LGTM!

File: docs/404.html
Comment (koheiw) [reply docs/404.html:1]: I removed docs, hoping that files in the folder will be generated automatically.

**Comment by koheiw** [comment]
@rcannood shall we submit v0.5 to CRAN? Is there anything we should do before that?
